app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -178,6 +180,26 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
         }
     }
 
+    @Override
+    public void onViewRecycled(RecyclerView.ViewHolder holder) {
+        super.onViewRecycled(holder);
+        if (holder.getItemViewType() == VIEWTYPE_MESSAGE)
+            markThisMessageAsRead((Message) getItem(holder.getAdapterPosition()));
+    }
+
+    private void markThisMessageAsRead(Message message) {
+        try {
+            int mID = message.getID();
+            if (!startedFromFilter && zulipApp.getPointer() < mID) {
+                Log.i("scrolling", "Now at " + mID);
```

Comment:
This log message is outdated -- it's now in a recycle function instead of a scrolling function.